File: Frontend/src/app/manutencao/models/usuario-operacao-retroativa.ts
import { Usuario } from "./usuario";

export class UsuarioOperacaoRetroativa {
    private constructor(
        codUsuario: string,
        limiteAlteracao: number,
        tipoProcesso : number,
        usuario : Usuario

    ) {
        this.codUsuario = codUsuario;
        this.limiteAlteracao = limiteAlteracao;
        this.tipoProcesso = tipoProcesso;
        this.usuario = usuario;
    }

    readonly codUsuario: string;
    readonly limiteAlteracao: number;
    readonly tipoProcesso : number;
    readonly usuario : Usuario;

    static fromObj(obj: UsuarioOperacaoRetroativaBack): UsuarioOperacaoRetroativa {
        return ({
            codUsuario: obj.codUsuario,
            limiteAlteracao: obj.limiteAlteracao,
            tipoProcesso: obj.tipoProcesso,
            usuario: obj.usuario
        });
    }
}

export class UsuarioOperacaoRetroativaBack {     
    readonly codUsuario: string;
    readonly limiteAlteracao: number;
    readonly tipoProcesso : number;
    readonly usuario : Usuario;
}

File: Backend/Perlink.Oi.Juridico.Infra/Entities/UsuarioOperacaoRetroativa.cs
using Perlink.Oi.Juridico.Infra.Seedwork;
using Perlink.Oi.Juridico.Infra.Seedwork.Notifying;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Perlink.Oi.Juridico.Infra.Entities
{
    public sealed class UsuarioOperacaoRetroativa : Notifiable, IEntity, INotifiable
    {
        private UsuarioOperacaoRetroativa()
        {
        }


        public static UsuarioOperacaoRetroativa Criar(string codUsuario, int limiteAlteracao, int tipoProcesso)
        {
            UsuarioOperacaoRetroativa operacao = new UsuarioOperacaoRetroativa();
            operacao.CodUsuario = codUsuario;
            operacao.LimiteAlteracao = limiteAlteracao;
            operacao.TipoProcesso = tipoProcesso;

            operacao.Validate();
            return operacao;
        }

        public void Atualizar(string codUsuario, int limiteAlteracao , int tipoProcesso)
        {
            CodUsuario = codUsuario;
            LimiteAlteracao = limiteAlteracao;
            TipoProcesso = tipoProcesso;

            Validate();
        }
        private void Validate()
        {
            if (string.IsNullOrEmpty(CodUsuario))
            {
                AddNotification(nameof(CodUsuario), "Campo Requerido.");
            }

            if (LimiteAlteracao <= 0)
            {
                AddNotification(nameof(CodUsuario), "Campo Requerido.");
            }

            if (TipoProcesso <= 0 )
            {
                AddNotification(nameof(CodUsuario), "Campo Requerido.");
            }
        }

        public string CodUsuario { get; private set; } 
        public int LimiteAlteracao { get; private set; }
        public int TipoProcesso { get; private set; }
        public Usuario Usuario { get; private set; }
    }
}